pkg/networkutils/network.go

```diff
@@ -116,30 +122,33 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDR *net.IPNet, primaryAddr *net.IP)
 		}
 	}
 
-	// Configure the kernel's reverse path filter check on eth0 for "loose" filtering.  This is required because
-	// - NodePorts are exposed on eth0
-	// - The kernel's RPF check happens after incoming packets to NodePorts are DNATted to the pod IP.
-	// - For pods assigned to secondary ENIs, the routing table includes source-based routing.  When the kernel does
-	//   the RPF check, it looks up the route using the pod IP as the source.
-	// - Thus, it finds the source-based route that leaves via the secondary ENI.
-	// - In "strict" mode, the RPF check fails because the return path uses a different interface to the incoming
-	//   packet.  In "loose" mode, the check passes because some route was found.
-	f, err := os.OpenFile("/proc/sys/net/ipv4/conf/eth0/rp_filter", os.O_WRONLY, 0644)
-	if err != nil {
-		return errors.Wrapf(err, "failed to configure eth0 RPF check")
-	}
-	defer f.Close()
-	const rpFilterLoose = "2"
-	_, err = f.WriteString(rpFilterLoose)
-	if err != nil {
-		return errors.Wrapf(err, "failed to configure eth0 RPF check")
+	if nodePortSupportEnabled {
```

Comment:
if NOT enabled, shall we revert this back to RPF?

Reply:
I was a little wary of doing that in case it was the user who set the value differently rather than the CNI plugin but it'd be easy to make that change if you think it's the right thing to do.

Reply:
I am fine with current approach.  For debugging field problems, can you add following
* add a new command, e,g `curl http://localhost:61678/env-settings` which dumps out current setting for RPF, CNI_CONNMARKS 
* add this new command into /opt/cni/aws-cni-support.sh
* also modify /opt/cni/aws-cni-support.sh to save a copy of /proc/sys/net/ipv4/conf/eth0/rp_filter setting inside aws-cni-support.tar.gz